### PR TITLE
Add Cosine 1D model

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

As part of some work I was doing, I found it convenient explicitly create a 1D cosine model formulated in the same fashion as the `Sine1D` model. Note that one could use the `Sine1D` model with a modified phase (in this formulation add 1/4 to the phase if was cosine). However, doing so can become tedious and/or cause user errors. Introducing a `Cosine1D` model helps to mitigate this by making the trigonometric model the user wishes to use more explicit.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
Tagging @perrygreenfield.

File: astropy/modeling/functional_models.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
            'Const2D', 'Ellipse2D', 'Disk2D', 'Gaussian1D', 'Gaussian2D',
            'Linear1D', 'Lorentz1D', 'RickerWavelet1D', 'RickerWavelet2D',
            'RedshiftScaleFactor', 'Multiply', 'Planar2D', 'Scale',
-           'Sersic1D', 'Sersic2D', 'Shift', 'Sine1D', 'Trapezoid1D',
+           'Sersic1D', 'Sersic2D', 'Shift', 'Sine1D', 'Cosine1D', 'Trapezoid1D',
```
Comment (perrygreenfield) [inline astropy/modeling/functional_models.py:19]: If we are going this route, it makes me wonder if we couldn't be complete with adding Tangent1D, and the inverse functions (though on the inverse, I guess sin.inverse() may be good enough; though looking at the code, am I mistaken in seeing that there is no inverse defined? If so, that seems like an omission--and yes, the inverse is not unique, but there are standard assumptions about which value to return).

**Review by perrygreenfield — APPROVED** [review]
LGTM

File: astropy/modeling/functional_models.py (hunk 1 above)
Comment (WilliamJamieson) [reply astropy/modeling/functional_models.py:19]: This has been done in PR #12185.

**Pull request merged**